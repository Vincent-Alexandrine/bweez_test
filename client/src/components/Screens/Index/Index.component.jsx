import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ErrorMessage from 'Atoms/ErrorMessage';
import Currencies from 'Molecules/Currencies';
import Header from 'Molecules/Header';
import Styled from './Index.styled.js';

class Index extends Component {
  static defaultProps = {

  };

  static propTypes = {
    match: PropTypes.shape({
      url: PropTypes.string,
    }),
    currencies: PropTypes.arrayOf(PropTypes.object),
    err: PropTypes.number,
    getCurrencies: PropTypes.func,
    getCurrency: PropTypes.func,
  };

  timeout = null;

  componentDidMount() {
    this.props.getCurrencies();
  }

  getCurrency = ({ target }) => {
    const { value } = target;
    const { getCurrency } = this.props;
    if (this.timeout)
      clearTimeout(this.timeout);
    this.timeout = setTimeout(() => getCurrency(value), 300);
  }

  render() {
    const { currencies, err } = this.props;
    return (
      <Styled>
        <Header name="header" getCurrency={this.getCurrency} />
        <main>
        {(err && (
          <ErrorMessage err={err} />
        )) || (
          <Currencies items={currencies} />
        )}
        </main>
      </Styled>
    );
  }
}

export default Index;
