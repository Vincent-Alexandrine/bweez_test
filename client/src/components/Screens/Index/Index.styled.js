import styled from 'styled-components';
import media from 'Styles/methods/media';

export default styled.div`
  display: grid;
  grid-template-rows: 15rem auto;
  grid-row-gap: 1rem;
  grid-template-columns: 12rem auto 12rem;
  grid-template-areas:
    ". header ."
    ". main .";

  ${media.tablet`
    grid-template-rows: 10rem auto;
    grid-template-columns: 9% auto 9%;
  `}
  main {
    grid-area: main;
  }

  [name="header"] {
    grid-area: header;
  }
`;
