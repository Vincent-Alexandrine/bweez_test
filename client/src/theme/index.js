import ThemeService from './theme.service.js';

export default ThemeService;
