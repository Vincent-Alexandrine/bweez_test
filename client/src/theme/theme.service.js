/*
  getTheme(string) -> theme
  addTheme(string, theme) -> void || true || promise
  removeTheme(string) -> void || true || promise

  validateTheme(theme) -> boolean
*/

const themesMap = new Map();

class ThemeService {
  static instance;

  constructor() {
    if (ThemeService.instance)
      return ThemeService.instance;

    ThemeService.instance = this;
  }

  get(name) {
    return themesMap.get(name);
  }

  add({ name, ...theme }) {
    themesMap.set(name, theme);
  }

  names() {
    return [...themesMap.keys()];
  }

  remove(name) {
    return themesMap.delete(name);
  }
}

const singleton = new ThemeService();

export default singleton;
