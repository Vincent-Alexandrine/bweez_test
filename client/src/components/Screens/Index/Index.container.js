import ReduxWrapper from 'Src/utilities/methods/redux.router.wrapper.js';
import { fetchCurrencies, fetchCurrency } from 'Actions/currencies.actions.js';
import Index from './Index.component.jsx';

// define Component's props from Store
// const mapStateToProps = (state) => ({
//  current: DefineCurrent(state.status, state.smt),   <-- pass props "current" to Component
// });

const mapStateToProps = (state) => ({
  currencies: state.currencies.items,
  isFetching: state.currencies.isFetching,
  err: state.currencies.err,
});

// define Component's cb as props, triggering "dispatch"
// const mapDispatchToProps = (dispatch) => ({
//   onSelect: id => dispatch(selectSmt(id)),   <-- pass props "onSelect" to Component
// });

const mapDispatchToProps = (dispatch) => ({
  getCurrencies: () => dispatch(fetchCurrencies()),
  getCurrency: (currency) => dispatch(fetchCurrency(currency)),
});

export default ReduxWrapper(
  mapStateToProps,
  mapDispatchToProps
)(Index);
