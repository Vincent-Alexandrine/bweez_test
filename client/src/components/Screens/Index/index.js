import Index from './Index.container.js';

export default Index;
