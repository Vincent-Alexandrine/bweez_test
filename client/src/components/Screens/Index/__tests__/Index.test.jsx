import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import 'jest-styled-components';
import theme from 'Styles/vManuscript';

import Index from '../Index.component.jsx';
import Styled from '../Index.styled.js';

const currencies = [];
const getCurrencies = () => {};
const getCurrency = () => {};

const props = {currencies, getCurrencies, getCurrency };

test('Index component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Index theme={theme} {...props} />
        </MemoryRouter>
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Index styled component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Styled />
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
