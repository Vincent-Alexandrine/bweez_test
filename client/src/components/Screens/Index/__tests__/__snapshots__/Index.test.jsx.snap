// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index component should match snapshot 1`] = `
.c2 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(20rem,1fr));
  grid-auto-rows: 1fr;
  grid-gap: 1rem;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c2::before {
  content: '';
  width: 0;
  padding-bottom: 100%;
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

.c2 > *:first-child {
  grid-row: 1 / 1;
  grid-column: 1 / 1;
}

.c1 {
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 [type="search"] {
  border-width: 0;
  padding: .5rem .5rem .5rem 1rem;
  background: #e6e6e6;
  border-radius: 2.5rem;
  -webkit-transition: all .1s ease-out;
  transition: all .1s ease-out;
  border-style: solid;
}

.c1 [type="search"]:focus {
  outline: none;
  background: transparent;
  box-shadow: inset 3px -2px 4px #cccccc, inset 0px 1px 3px #cccccc, inset 3px 1px 4px #cccccc;
}

.c0 {
  display: grid;
  grid-template-rows: 15rem auto;
  grid-row-gap: 1rem;
  grid-template-columns: 12rem auto 12rem;
  grid-template-areas: ". header ." ". main .";
}

.c0 main {
  grid-area: main;
}

.c0 [name="header"] {
  grid-area: header;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: repeat(auto-fill,minmax(30%,1fr));
  }
}

@media (max-width:576px) {
  .c2 {
    grid-template-columns: repeat(auto-fill,minmax(45%,1fr));
  }
}

@media (max-width:576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (max-width:768px) {
  .c0 {
    grid-template-rows: 10rem auto;
    grid-template-columns: 9% auto 9%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    name="header"
  >
    <h1>
      title
    </h1>
    <input
      onChange={[Function]}
      type="search"
    />
  </div>
  <main>
    <ul
      className="c2"
    />
  </main>
</div>
`;

exports[`Index styled component should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-rows: 15rem auto;
  grid-row-gap: 1rem;
  grid-template-columns: 12rem auto 12rem;
  grid-template-areas: ". header ." ". main .";
}

.c0 main {
  grid-area: main;
}

.c0 [name="header"] {
  grid-area: header;
}

@media (max-width:768px) {
  .c0 {
    grid-template-rows: 10rem auto;
    grid-template-columns: 9% auto 9%;
  }
}

<div
  className="c0"
/>
`;
